domain/lib/src/usecases/download_file/download_file_exception.dart

```diff
@@ -29,18 +29,34 @@
 //  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
 //  the Additional Terms applicable to LinShare software.
 
-class DeviceNotSupportedException implements Exception {
-  String notSupported = 'This device is not supported, please try on Android or iOS';
-}
+import 'package:equatable/equatable.dart';
 
-class DownloadFileException implements Exception {
+abstract class DownloadFileException extends Equatable implements Exception {
   final String message;
 
   DownloadFileException(this.message);
+
+  @override
+  String toString() => message;
+
+  @override
+  List<Object> get props => [message];
 }
 
-class CancelDownloadFileException implements Exception {
-  final String message;
+class CommonDownloadFileException extends DownloadFileException {
+  CommonDownloadFileException(message) : super(message);
+
+  @override
+  List<Object> get props => [message];
+}
+
+class CancelDownloadFileException extends DownloadFileException {
+  CancelDownloadFileException(cancelMessage) : super(cancelMessage);
+
+  @override
+  List<Object> get props => [message];
+}
 
-  CancelDownloadFileException(this.message);
+class DeviceNotSupportedException extends DownloadFileException {
```

Comment:
where you use this exception?

Reply:
I just refactored extensions so they can be equatable, this one already existed before